### ISSUE TITLE
Style/Copyright cop autocorrecting without hashtag/comment

### ISSUE THREAD

**Issue by Falciron** [rubocop/rubocop#15130]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

Incorrect autocorrect

### Cop name

Style/Copyright

### Steps to reproduce the problem

1. Create a file `example.rb` with a single line of `puts 'Hello world'`
2. Add the .rubocop.yml configuration file, as below. (Try both configs.)
3. Run `rubocop -A example.rb`.
4. Review terminal output and code file.

### Expected behavior

The first configuration file should insert the AutocorrectNotice text above the line of code, with an octothorpe and a space (# ) prefacing it and not result in further warnings.

The second configuration file should insert the AutocorrectNotice text above the line of code and not result in further warnings.

### Actual behavior

The first configuration file will insert the AutocorrectNotice text above the line of code, without a # prefacing it, resulting in invalid Ruby code. This results in other cops reporting issues, if enabled, and code that doesn't run.

The second configuration file will result in a warning, saying `Style/Copyright: AutocorrectNotice '# Copyright (c) 2026 My Name' must match Notice /^# Copyright (\(c\) )?2[0-9]{3} .+/ (from file: ...`.

### RuboCop configuration

```yaml
There are two configs worth inspecting - the first omits the # from the config (and relies upon the autocorrect to place it), while the second includes the # in the config and fails to autocorrect with a warning due to not matching the regex.

AllCops:
  DisabledByDefault: true

Style/Copyright:
  Enabled: true
  Notice: '^Copyright (\(c\) )?2[0-9]{3} .+'
  AutocorrectNotice: 'Copyright (c) 2026 My Name'

--------

AllCops:
  DisabledByDefault: true

Style/Copyright:
  Enabled: true
  Notice: '^# Copyright (\(c\) )?2[0-9]{3} .+'
  AutocorrectNotice: '# Copyright (c) 2026 My Name'
```

### RuboCop version

```shell
$ rubocop -V
1.86.1 (using Parser 3.3.11.1, rubocop-ast 1.49.1, analyzing as Ruby 2.7, running on ruby 3.4.6) [arm64-darwin25]
```

**Comment by Falciron** [comment]
This issue is similar to #2068, but as it's been over a decade, I believe it's fair to open up a new issue.
